File: src/main/java/com/nobility/downloader/downloads/DownloadUpdater.java
package com.nobility.downloader.downloads;

import com.nobility.downloader.Model;

public class DownloadUpdater implements Runnable {

    private final Model model;
    private final Download download;
    private boolean running = true;

    public DownloadUpdater(final Model model, final Download download) {
        this.model = model;
        this.download = download;
    }

    @Override
    public void run() {
        while (running) {
            model.updateDownloadProgress(download);
            try {
                Thread.sleep(200);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }

    public void setRunning(boolean running) {
        this.running = running;
    }
}

File: src/main/java/com/nobility/downloader/downloads/DownloadUpdater.kt
package com.nobility.downloader.downloads

import com.nobility.downloader.Model
import com.nobility.downloader.entities.Download
import kotlinx.coroutines.delay

class DownloadUpdater(private val model: Model, private val download: Download) {

    private var running = true

    suspend fun run() {
        while (running) {
            model.updateDownloadProgress(download)
            delay(500)
        }
        model.updateDownloadProgress(download)
    }

    fun setRunning(running: Boolean) {
        this.running = running
    }
}